spec/file_translation/fts_language_detection.yaml

```diff
@@ -0,0 +1,167 @@
+x-paths:
+
+  language_detection_action:
+    post:
+      summary: Detect file language
+      description: >-
+        Trigger detection of source language of earlier uploaded file
+      tags:
+        - File Machine Translation
+      operationId: detectFileSourceLanguage
+      produces:
+        - application/json
+      parameters:
+        - $ref: './fts_common.yaml#/components/parameters/accountUidPathParameter'
+        - $ref: './fts_common.yaml#/components/parameters/fileUidPathParameter'
+      responses:
+        '202':
+          content:
+            application/json:
+              examples:
+                response:
+                  value:
+                    response:
+                      code: ACCEPTED
+                      data:
+                        languageDetectionUid: "h5dhjbg1ganp"
+              schema:
+                $ref: '#/components/schemas/LanguageDetectionResponse'
+          description: OK
+        '400':
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error400Response'
+              examples:
+                response:
+                  value:
+                    response:
+                      code: VALIDATION_ERROR
+                      errors:
+                        key: not.found
+                        message: >-
+                          fileUid h5dhjbg1ganp not found
+                        details:
+                          field: "fileUid"
+
+        '401':
+          $ref: '#/components/responses/Error401ResponseDefinition'
+        '429':
+          $ref: '#/components/responses/Error429ResponseDefinition'
+        '500':
+          $ref: '#/components/responses/Error500ResponseDefinition'
+      x-code-samples:
+        - lang: curl
+          source: >
+            curl -X POST -H "Authorization: Bearer $smartlingToken" 
+            https://api.smartling.com/file-translations-api/v2/accounts/$smartlingAccountId/files/$smartlingFtsFileUid/language-detection
+
+  language_detection_status:
+    get:
+      summary: Language detection progress
+      description: |-
+        Fetch language detection progress/result.
```

Comment:
Get progress for an ongoing source language detection action. Result of language detection is propagated to response as soon as state changes to COMPLETED.

Reply:
fixed